jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/core/jsonp/JsonpDiagnosticCollector.java

```diff
@@ -32,7 +42,39 @@ public void completeDiagnostic(Diagnostic diagnostic) {
 
     @Override
     public void collectDiagnostics(ICompilationUnit unit, List<Diagnostic> diagnostics) {
+        if (unit == null) {
+            return;
+        }
+        List<MethodInvocation> allMethodInvocations = ASTUtils.getMethodInvocations(unit);
+        List<MethodInvocation> createPointerInvocations = allMethodInvocations.stream()
+                .filter(this::isCreatePointerInvocation).collect(Collectors.toList());
```

Comment:
Suggestion (I'll leave to it to you on whether you prefer to leave it as is): consider using a for loop here for readability